### ISSUE TITLE
ccxt proxy can not work

### ISSUE THREAD

**Issue by skyold** [ccxt/ccxt#15813]
OS: MacOS 13.0.1
    Programming Language version: Typescript with node 18.11.7
    CCXT version: 2.0.77

Code

**Code1: Not work!  My code** 
RequestTimeout: binance GET https://api.binance.com/api/v3/exchangeInfo request timed out (10000 ms) 

 ```
  const proxy = 'https://127.0.0.1:10077' 
  const agent = new HttpsProxyAgent (proxy)
  const binance:ccxt.Exchange = new ccxt.binance ({agent})
  const binace_markets = await binance.fetchMarkets()
  console.log(binace_markets)
```

**Code2: Working fine ! the code to make sure environment is ok:** 
 ```
  const proxy = 'https://127.0.0.1:10077' 
  const agent = new HttpsProxyAgent (proxy)
  var endpoint = 'https://api.binance.com/api/v3/exchangeInfo'
  console.log('attempting to GET %j', endpoint);
  var options = url.parse(endpoint);
  options.agent = agent;
  https.get(options, function (res:any) {
      console.log('"response" event!', res.headers);
      res.pipe(process.stdout);
  });
```

**Code3:Not Work!  the previous answer with similar issue**
   ```
 const ccxt = require ('ccxt')
        , HttpsProxyAgent = require ('https-proxy-agent')
        , http = require('https')
        , agent = new http.Agent()
        // a workaround to force Agent-type on the https-proxy-agent
        , _ = Object.assign (agent, new HttpsProxyAgent ('https://127.0.0.1:10077'))
        , exchange = new ccxt.binance ({ agent })
    console.log (await exchange.fetchMarkets ()) 





```

**Comment by sc0Vu** [comment]
Hi @skyold 

Have you checked your proxy server? It did work for me (I also test socks proxy).

**Comment by skyold** [comment]
Thanks @sc0Vu .   the second code working fine on my side, so looks like the proxy is somehow working. 
I also try the following curl command it is work as well 
`curl  https://api.binance.com/api/v3/exchangeInfo -x http://127.0.0.1:10077`

**Comment by skyold** [comment]
Attach the full error 

```
(node:6117) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
/node_modules/ccxt/js/base/Exchange.js:1814
        return await this.fetch (request['url'], request['method'], request['headers'], request['body']);
               ^
RequestTimeout: binance GET https://api.binance.com/api/v3/exchangeInfo request timed out (10000 ms)
    at /node_modules/ccxt/js/base/Exchange.js:436:27
    at async binance.fetch2 (/node_modules/ccxt/js/base/Exchange.js:1814:16)
    at async binance.request (/node_modules/ccxt/js/binance.js:6006:26)
    at async binance.fetchMarkets (/node_modules/ccxt/js/binance.js:1467:26)
```

**Comment by frosty00** [comment]
I have fixed this issue in #15899
